org/qmckl_sherman_morrison_woodbury.org

```diff
@@ -696,12 +833,59 @@ qmckl_exit_code qmckl_sherman_morrison_splitting_c(const qmckl_context context,
 
    #+CALL: generate_c_interface(table=qmckl_sherman_morrison_splitting_args,rettyp=get_value("FRetType"),fname=get_value("Name"))
 
+   #+RESULTS:
+   #+begin_src f90 :tangle (eval f) :comments org :exports none
+   integer(c_int32_t) function qmckl_sherman_morrison_splitting &
+       (context, Dim, N_updates, Updates, Updates_index, breakdown, Slater_inv) &
+       bind(C) result(info)
+
+     use, intrinsic :: iso_c_binding
+     implicit none
+
+     integer (c_int64_t) , intent(in) , value :: context
+     integer (c_int64_t) , intent(in)         :: Dim
+     integer (c_int64_t) , intent(in)         :: N_updates
+     real    (c_double ) , intent(in)         :: Updates(N_updates*Dim)
+     integer (c_int64_t) , intent(in)         :: Updates_index(N_updates)
+     real    (c_double ) , intent(in)         :: breakdown
+     real    (c_double ) , intent(inout)      :: Slater_inv(Dim*Dim)
+
+     integer(c_int32_t), external :: qmckl_sherman_morrison_splitting_c
+     info = qmckl_sherman_morrison_splitting_c &
+	    (context, Dim, N_updates, Updates, Updates_index, breakdown, Slater_inv)
+
+   end function qmckl_sherman_morrison_splitting
+   #+end_src
+
    #+CALL: generate_f_interface(table=qmckl_sherman_morrison_splitting_args,rettyp=get_value("FRetType"),fname=get_value("Name"))
 
+   #+RESULTS:
+   #+begin_src f90 :tangle (eval fh_func) :comments org :exports none
+   interface
+     integer(c_int32_t) function qmckl_sherman_morrison_splitting &
+	 (context, Dim, N_updates, Updates, Updates_index, breakdown, Slater_inv) &
+	 bind(C)
+       use, intrinsic :: iso_c_binding
+       import
+       implicit none
+
+       integer (c_int64_t) , intent(in) , value :: context
+       integer (c_int64_t) , intent(in)         :: Dim
+       integer (c_int64_t) , intent(in)         :: N_updates
+       real    (c_double ) , intent(in)         :: Updates(N_updates*Dim)
+       integer (c_int64_t) , intent(in)         :: Updates_index(N_updates)
+       real    (c_double ) , intent(in)         :: breakdown
+       real    (c_double ) , intent(inout)      :: Slater_inv(Dim*Dim)
+
+     end function qmckl_sherman_morrison_splitting
+   end interface
+   #+end_src
+
 *** Test                                                           :noexport:
 
      #+begin_src c :tangle (eval c_test)
-rc = qmckl_sherman_morrison_splitting_c(context, Dim, N_updates3, Updates3, Updates_index3, breakdown, Slater_inv3_2);
+N_updates_p = &N_updates3;
```

Comment:
Shouldn't it be `N_updates3_p` ? Just wondering.

Reply:
Actually no. I am reusing the same pointer 'N_updates_p', reassigning it at each test.

Reply:
OK.